src/api/Elastic.Documentation.Api.Lambda/Dockerfile

```diff
@@ -5,33 +5,37 @@ ARG TARGETOS
 
 WORKDIR /app
 
-RUN rpm --import https://packages.microsoft.com/keys/microsoft.asc
-
-#RUN curl -o /etc/yum.repos.d/microsoft-prod.repo https://packages.microsoft.com/config/fedora/41/prod.repo
-
-RUN dnf update -y
-#RUN dnf install -y dotnet-sdk-10.0
-RUN dnf install -y npm
-RUN dnf install -y git
-RUN dnf install -y clang
-
-## temporary see: https://github.com/amazonlinux/amazon-linux-2023/issues/1025
-RUN dnf install -y tar
-RUN dnf install -y findutils
-RUN curl -L https://dot.net/v1/dotnet-install.sh -o dotnet-install.sh
-RUN chmod +x ./dotnet-install.sh
-RUN ./dotnet-install.sh
-ENV PATH="$PATH:/root/.dotnet"
-ENV DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1
-# /end temporary
+# Set environment variables early (no layer cost)
+ENV PATH="$PATH:/root/.dotnet" \
+    DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1 \
+    DOTNET_NOLOGO=true \
+    DOTNET_CLI_TELEMETRY_OPTOUT=true
 
+# Install all system dependencies in a single layer
+RUN rpm --import https://packages.microsoft.com/keys/microsoft.asc && \
+    dnf update -y && \
+    dnf install -y npm git clang tar findutils && \
+    dnf clean all
+
+# Install .NET SDK based on global.json version
```

Comment:
I think we should leave in the comment that this might be temporary:

https://github.com/amazonlinux/amazon-linux-2023/issues/1025